uuid_source.go

```diff
@@ -0,0 +1,54 @@
+package uuid
+
+import (
+	"io"
+	"crypto/rand"
+)
+
+// A UuidSource holds a random number generator and generates UUIDs using it as its source.
+//
+// It is useful when a process need its own random number generator, 
+// e.g. when running some processes concurrently.
+type UuidSource struct {
+	rander io.Reader
+}
+
+// Creates a new UuidSource which holds its own random number generator.
+//
+// Calling NewSource with nil sets the random number generator to a default
+// generator.
+func NewSource(r io.Reader) UuidSource {
+	var uuidSource UuidSource
```

Comment:
return Source{render: r}

No reason to special case nil.  They can pass in rand.Reader if they want.